agent/engine/task_manager.go

```diff
@@ -352,12 +374,39 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 		mtask.UpdateMountPoints(container, event.Volumes)
 	}
 
-	mtask.engine.emitContainerEvent(mtask.Task, container, "")
+	mtask.emitContainerEvent(mtask.Task, container, "")
 	if mtask.UpdateStatus() {
 		llog.Debug("Container change also resulted in task change")
 		// If knownStatus changed, let it be known
-		mtask.engine.emitTaskEvent(mtask.Task, "")
+		mtask.emitTaskEvent(mtask.Task, "")
+	}
+}
+
+func (mtask *managedTask) emitTaskEvent(task *api.Task, reason string) {
+	event, err := api.NewTaskStateChangeEvent(task, reason)
+	if err != nil {
+		seelog.Debugf("Unable to create task state change event for task '%s': %v",
+			task.Arn, err)
+		return
+	}
+
+	seelog.Infof("Managed task: sending change event [%s]", event.String())
```

Comment:
readability: Can we get logging message parity between this `Infof` for `emitTaskEvent` and the one below for `emitContainerEvent`? We seem to have one that's info level before passing it to the channel, one after, and also the two messages are formatted differently. Also one has a `Debugf` level message before putting the event on the channel.